src/parse.rs

```diff
@@ -2580,14 +2577,15 @@ impl<'a> Parser<'a> {
 
     /// Returns the next event in a pre-order AST walk, along with its
     /// start and end offset in the source.
+    #[inline(always)]
```

Comment:
Would just `#[inline]` be a serious performance regression? Generally `always` is frowned on, and this is a pretty big function to inline.

Reply:
It's not serious, but it is measurable: around 2%. This call is very hot. The reason for forcing inlining it was that the regular (non source-mapped) iterator has to strip source information otherwise. If it is inlined, the source references need not even be formed.

I suspect it's not inlined because it's fairly large as you say and it's used by two distinct iterators. But these are mutually exclusive in use for any given parser. So while forcing inlining here may increase code size a bit, it shouldn't hurt instruction caches.

Reply:
I have left a comment on it with the current considerations. We can revisit it later if when we find a better solution.